### ISSUE TITLE
Point visualization and vispy 0.9.6

### ISSUE THREAD

**Issue by leoguignard** [napari/napari#4186]
## 🐛 Bug

vispy 0.9.6 does not allow to represent points alone as previous versions did.
Reverting to vispy 0.6.6 and napari 0.4.9 solves the issue

## To Reproduce

```shell
pip install napari
```

```python
import napari
import vispy
import numpy as np

print(f'vispy version: {vispy.__version__}')
print(f'napari version: {napari.__version__}')

nb_points = 20
x = np.arange(nb_points)
y = np.arange(nb_points)
z = np.arange(nb_points)
points = np.array(np.meshgrid(x, y, z)).T.reshape(-1, 3)
viewer = napari.view_points(points, size=.2)
```
output:
```
vispy version: 0.9.6
napari version: 0.4.14
```

## Expected behavior

Display the points

## Environment

When it is not working:
> napari: 0.4.14
> Platform: macOS-10.16-x86_64-i386-64bit
> System: MacOS 11.4
> Python: 3.9.7 (default, Sep 16 2021, 08:50:36) [Clang 10.0.0 ]
> Qt: 5.15.2
> PyQt5: 5.15.6
> NumPy: 1.21.2
> SciPy: 1.7.3
> Dask: 2021.10.0
> VisPy: 0.9.6
> 
> OpenGL:
> - GL version: 2.1 ATI-4.5.14
> - MAX_TEXTURE_SIZE: 16384
> 
> Screens:
> - screen 1: resolution 1680x1050, scale 2.0
> 
> Plugins:
> - console: 0.0.4
> - scikit-image: 0.4.14
> - svg: 0.1.6

When it is working (differences in bold):
> napari: **0.4.9**
> Platform: macOS-10.16-x86_64-i386-64bit
> System: MacOS 11.4
> Python: 3.9.7 (default, Sep 16 2021, 08:50:36) [Clang 10.0.0 ]
> Qt: 5.15.2
> PyQt5: 5.15.6
> NumPy: 1.21.2
> SciPy: 1.7.3
> Dask: 2021.10.0
> VisPy: **0.6.6**
> 
> OpenGL:
> - GL version: 2.1 ATI-4.5.14
> - MAX_TEXTURE_SIZE: 16384
> 
> Screens:
> - screen 1: resolution 1680x1050, scale 2.0
> 
> Plugins:
> - console: 0.0.4
> - scikit-image: 0.4.9
> - svg: 0.1.6

**Comment by kevinyamauchi** [comment]
Thank you for the clear bug report, @leoguignard !

I am not sure if you just left it off of the reproducible example, but when running napari from a script, you need to add `napari.run()` at the end to start the event loop.

```python
import napari
import vispy
import numpy as np

print(f'vispy version: {vispy.__version__}')
print(f'napari version: {napari.__version__}')

nb_points = 20
x = np.arange(nb_points)
y = np.arange(nb_points)
z = np.arange(nb_points)
points = np.array(np.meshgrid(x, y, z)).T.reshape(-1, 3)
viewer = napari.view_points(points, size=.2)

napari.run()
```

Even after adding the `napari.run()` to the end, the points are not visible in the viewer. However, they are visible if I make the points bigger (e.g., set the size to 10). Is that the case for you?

@brisvag , do you think this has to do with the point size and edge width? Or is this the weird behavior we've seen when points are too close together (points are one unit apart)?

**Comment by brisvag** [comment]
I'll look better later (on the phone now), but just from reading the issue this gives me the "extent issue" vibes. Are dims sliders visible?

**Comment by leoguignard** [comment]
Hello, thanks for the quick answer.

I did not had the `napari.run()` since I was running napari from a notebook. I did not specify that part, I did not think it could have an impact.

The `size` seems to have an impact indeed but maybe not the expected one. Also, having a higher size makes the visualization and interaction significantly slower.

Maybe a few quick few things that I forgot to mention in the original post, it's likely obvious to you but just in case it is not:
- I *had* to downgrade the napari from version `0.4.14` to `0.4.9` to be able to have it work with vispy version `0.6.6`
- I have not tried versions of napari other than `0.4.9` and `0.4.14`
- I have not tried versions of vispy other than `0.6.6` and `0.9.6`
- I am mainly pointing out that the behaviors between the two versions are different, I tagged it as a bug because I like the behavior of the older version. The new behavior might be intended though.

That being said, if it helps here are the screen cap of the outputs that I get from the two different versions (note that you can correctly see the points in the layer preview)
- `napari==0.4.14 vispy==0.9.6 size=.2`
<img width="600" alt="napari==0.4.14 vispy==0.9.6 size=.2" src="https://user-images.githubusercontent.com/31226251/156482567-31025bd7-0205-435a-99de-fcc4108ee41a.png">

- `napari==0.4.14 vispy==0.9.6 size=10`
<img width="600" alt="Screen Shot 2022-03-03 at 02 58 58" src="https://user-images.githubusercontent.com/31226251/156482779-647ec687-cf6a-43f2-a258-cf781f027be7.png">

- `napari==0.4.9 vispy==0.6.6 size=.2`
<img width="600" alt="Screen Shot 2022-03-03 at 02 46 35" src="https://user-images.githubusercontent.com/31226251/156482854-36b0d6e7-f966-4997-b9c3-89e5dc758a86.png">

- `napari==0.4.9 vispy==0.6.6 size=10`
<img width="600" alt="Screen Shot 2022-03-03 at 03 01 37" src="https://user-images.githubusercontent.com/31226251/156482888-3c0e6241-2fd5-47b6-b897-594102170187.png">

**Comment by brisvag** [comment]
Ok, I took a better look, @leoguignard. The issue should be solved already in main (which will be released with v0.4.15 next Tuesday!). The issue was cause by a long-standing problem with size, edge width and sensible defaults.

As for what fixes it in main: it's a change I introduced in #3999, which makes the `edge_width` of points relative to the size by default. This means that if you have `size=10` and `edge_width=0.2`, you actually get edge width `2`. On the flipside, on 0.4.14, if you have size `0.2` (like in your example), `edge_width` *still* remains the default (`1`, if I recall correctly), which is bigger than the size of the point. So you get points with an edge bigger than the face. If the edge is also black like the background: *boom*, invisible points :) The change in #3999 was specifically to avoid this kind of issue!

TL;DR: in the meantime, simply changing the `edge_width` to a smaller size should fix your problem, but you won't have to do it starting from `v0.4.15`

**Comment by kevinyamauchi** [comment]
Thanks, @brisvag ! I just pulled main and tested @leoguignard's  example and it looks good.

<img width="1204" alt="Screenshot 2022-03-03 at 10 55 06" src="https://user-images.githubusercontent.com/1120672/156540883-a7f72fa1-bb12-4d28-a0ca-521b3344037d.png">

Also, thanks for the clear question and documentation of the issue, @leoguignard - makes it much easier to debug! I will close this for now, but please feel free to re-open if pulling main or the forthcoming `0.4.15` release doesn't fix your issue.

**Comment by jni** [comment]
It might be helpful to note that you can test the 0.4.15 release candidate (0.4.15rc0) now with `pip install -U --pre napari`!

**Comment by leoguignard** [comment]
Perfect, thanks a lot for the quick answers!
